services/src/main/java/org/fao/geonet/api/AllRequestsInterceptor.java

```diff
@@ -23,49 +23,42 @@
 
 package org.fao.geonet.api;
 
+import jeeves.constants.Jeeves;
+import jeeves.server.UserSession;
 import org.apache.commons.lang.StringUtils;
 import org.fao.geonet.utils.Log;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
 
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
-
-import jeeves.constants.Jeeves;
-import jeeves.server.UserSession;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * In charge of creating a new {@link UserSession} if not existing. Avoid to create any sessions for
- * crawlers.
+ * In charge of creating a new {@link UserSession} if not existing.
+ * Avoid to create any sessions for crawlers.
  */
 public class AllRequestsInterceptor extends HandlerInterceptorAdapter {
+
     /**
      * List of bots to avoid.
      */
-    public static final String BOT_REGEXP = ".*(bot|crawler|baiduspider|80legs|ia_archiver|"
-        + "voyager|yahoo! slurp|mediapartners-google).*";
+    @Value("${bot.regexpFilter:.*(bot|crawler|baiduspider|80legs|ia_archiver|voyager|yahoo! slurp|mediapartners-google).*}")
+    public String botRegexpFilter = "";
 
-    private static final Pattern regex = Pattern.compile(BOT_REGEXP, Pattern.CASE_INSENSITIVE);
+    private Pattern regex = Pattern.compile(botRegexpFilter, Pattern.CASE_INSENSITIVE);
```

Comment:
Are there any cases where this actually works?
@Value are only parse after the class has been initialized so the pattern will always be based on the initial value of botRegexpFilter which is ""
I believe this should be done in a postconstruct